varc_core/systems/linux.py

```diff
@@ -107,9 +108,21 @@ def dump_processes(self) -> None:
                             for map in maps:
                                 page_start = map[0]
                                 page_len = map[1] - map[0]
-                                mem_page_content = self.read_bytes(pid, page_start, page_len)
-                                if mem_page_content:
-                                    tmpfile.write(mem_page_content)
+                                if page_len > self._MAX_VIRTUAL_PAGE_CHUNK:
+                                    sub_chunk_count, final_chunk_size = divmod(page_len, self._MAX_VIRTUAL_PAGE_CHUNK)
+                                    page_len = int(page_len / sub_chunk_count)
+                                    for sc in range(0, sub_chunk_count):
```

Comment:
Rather than doing a loop followed by the final write to get the last (which introduces some duplication) you can do with a single loop. Something like:

```
for chunk_start in range(0, page_len, self._MAX_VIRTUAL_PAGE_CHUNK):
  mem_page_content = self.read_bytes(pid, chunk_start, min(chunk_start + self._MAX_VIRTUAL_PAGE_CHUNK, page_len)
  if mem_page_content:
     tmpfile.write(mem_page_content)
```